go 1.24 panic: `invalid table access`

Reported in https://github.com/gitleaks/gitleaks/issues/1751#issue-2847350166; I haven't been able to reproduce this yet.

```
panic: wasm error: invalid table access
        wasm stack trace:
                .$27(i32,i32,i32,i32)
                .$398(i32)
                .$430(i32,i32)
                .$428(i32,i32,i32)
                .$474(i32)
                .$652(i32,i32,i32)
                .$666(i32,i32,i32) i32

goroutine 1 [running]:
github.com/wasilibs/go-re2/internal.newRE(0xc0013df680, {0x5885e0?, 0xc001b164f8?}, {0x39?, 0x3c?, 0x47?, 0x0?})
        /go/pkg/mod/github.com/wasilibs/go-re2@v1.8.0/internal/re2_wazero.go:289 +0x269
github.com/wasilibs/go-re2/internal.Compile({0xc0005885e0, 0x20}, {0xd5?, 0xa3?, 0x6a?, 0x76?})
        /go/pkg/mod/github.com/wasilibs/go-re2@v1.8.0/internal/re2.go:67 +0x165
github.com/wasilibs/go-re2.Compile(...)
        /go/pkg/mod/github.com/wasilibs/go-re2@v1.8.0/re2.go:46
github.com/wasilibs/go-re2.MustCompile({0xc0005885e0, 0x20})
        /go/pkg/mod/github.com/wasilibs/go-re2@v1.8.0/re2.go:76 +0x2e
github.com/zricethezav/gitleaks/v8/regexp.MustCompile(...)
        /pwd/gitleaks/regexp/wasilibs_regex.go:14
```

--- a/go.mod
+++ b/go.mod
@@ -3,7 +3,7 @@ module github.com/wasilibs/go-re2
 go 1.22.0
 
 require (
-	github.com/tetratelabs/wazero v1.8.2
+	github.com/tetratelabs/wazero v1.9.0
 	github.com/wasilibs/wazero-helpers v0.0.0-20240620070341-3dff1577cd52
 )
 

